packages/react-interactions/events/src/dom/create-event-handle/__tests__/useFocusWithin-test.internal.js

```diff
@@ -528,14 +541,17 @@ describe.each(table)(`useFocus`, hasPointerEvents => {
 
       suspend = true;
       await act(() => {
-        root.render(<Component />);
+        root2.render(<Component />);
       });
       expect(container2.innerHTML).toBe(
         '<div><input style="display: none;">Loading...</div>',
       );
       expect(onBeforeBlurWithin).toHaveBeenCalledTimes(1);
       expect(onAfterBlurWithin).toHaveBeenCalledTimes(1);
-      resolve();
+      await act(() => {
+        suspend = false;
+        resolve();
+      });
```

Comment:
Matches other testing patterns e.g. https://github.com/facebook/react/blob/a8875eab7f78a453d22370d1061a8bb3cd672b9d/packages/react-dom/src/__tests__/ReactTestUtilsAct-test.js#L773-L777

Without also ensuring we don't throw the promise, we end up in an infinite loop and ultimately crash with `RangeError: Potential infinite loop: exceeded 1500 iterations.`